app/src/main/java/com/nextlevelstudy/StudentEnrollmentApp.java

```diff
@@ -0,0 +1,36 @@
+package com.nextlevelstudy;
+
+import android.app.Activity;
+import android.app.Application;
+
+import com.nextlevelstudy.di.AppInjector;
+
+import javax.inject.Inject;
+
+import dagger.android.AndroidInjector;
+import dagger.android.DispatchingAndroidInjector;
+import dagger.android.HasActivityInjector;
+import timber.log.Timber;
+
+public class StudentEnrollmentApp extends Application implements HasActivityInjector {
+
+  @Inject
+  DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;
+
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+
+    if (BuildConfig.DEBUG) {
+      Timber.plant(new Timber.DebugTree());
+    }
+
+    AppInjector.init(this);
```

Comment:
AppInjector is nice, but now Dagger handles everything using DaggerApplication, DaggerActivity, DaggerFragment, DaggerService etc. 
Check out this article (one of many): https://android.jlelse.eu/new-android-injector-with-dagger-2-part-3-fe3924df6a89